subgraphs/_reference_/src/common/utils/numbers.ts

```diff
@@ -8,6 +8,14 @@ export function bigIntToBigDecimal(quantity: BigInt, decimals: i32 = 18): BigDec
   );
 }
 
+export function powerBigDecimal(base: BigDecimal, power: i32): BigDecimal {
```

Comment:
I believe pow is optimized so that it runs faster than the for-loop-times below

Here're how I do it
```ts
// Formula: check out "Calculating the APY Using Rate Per Block" section https://compound.finance/docs
function convertRatePerBlockToAPY(ratePerBlock: BigInt): BigDecimal {
  let a = mantissaFactorBI
  let b = ratePerBlock.times(BLOCKS_PER_DAY).plus(mantissaFactorBI)
  return pow(a, b, 365).minus(BIGDECIMAL_ONE).times(BIGDECIMAL_HUNDRED)
}

// (a/b)^n
function pow(a: BigInt, b: BigInt, n: i32): BigDecimal {
  return a.pow(n).toBigDecimal().div(b.pow(n).toBigDecimal())
}
```

Reply:
yeah, powering can generally be done in logarithmic time: https://stackoverflow.com/questions/101439/the-most-efficient-way-to-implement-an-integer-based-power-function-powint-int

Reply:
My struggle here is that blocksPerDay is a BigDecimal -- any thoughts on how to tackle this?

Reply:
@dmelotik cast blocks_per_day into a bigdecimal then 🤔 i believe that's acceptable

Reply:
@0xbe1 do you mean a BigInt?  - blocksPerDay is already a BigDecimal.

Reply:
yeah i mean BigInt

Reply:
turns out the approach I propose doesn't work because BigInt.pow(n) expects n between 0 and 255, while 365 is out of range 🤒

Reply:
@0xbe1 see my latest solution and lmk your thoughts

Reply:
we could keep the `pow` approach after graphprotocol fix their gas estimation on `pow`